ibis_substrait/tests/integration/test_pyarrow.py

```diff
@@ -0,0 +1,101 @@
+import ibis
+import pytest
+
+from ibis_substrait.compiler.core import SubstraitCompiler
+
+compiler = SubstraitCompiler()
+
+
+def get_table_provider(tbl):
+    """Create a table_provider that always returns tbl"""
+
+    def table_provider(names, schema):
+        return tbl
+
+    return table_provider
+
+
+def to_ibis_table(arrow_table, table_name="t1"):
+    """Create ibis Table from pyarrow Table"""
+    return ibis.table(
+        zip(arrow_table.schema.names, [str(type) for type in arrow_table.schema.types]),
+        name=table_name,
+    )
+
+
+def run_query(expression, tbl, pa, substrait):
+    compiled = compiler.compile(expression)
+    query_bytes = compiled.SerializeToString()
+    result = substrait.run_query(
+        pa.py_buffer(query_bytes), table_provider=get_table_provider(tbl)
+    )
+
+    results = result.read_all()
+    assert type(results) == pa.lib.Table
+
+    return results
+
+
+# WIP: I'm not sure how best to parameterize the query too without a ref to the Table
```

Comment:
Yeah, so I think if we do the top-level importorskip we can have the `output` be the expected pyarrow objects without worrying about import headaches

Reply:
ah yep. that's a lot easier